diffsims/utils/gridding_utils.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The diffsims developers
+#
+# This file is part of diffsims.
+#
+# diffsims is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# diffsims is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with diffsims.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Helper functions for gridding
+"""
+
+import numpy as np
+from itertools import product
+from transforms3d.euler import axangle2euler, euler2axangle, euler2mat
+from transforms3d.quaternions import quat2axangle, axangle2quat, mat2quat, qmult
+from diffsims.utils.rotation_conversion_utils import *
+
+
+def vectorised_qmult(q1, qdata):
+    """ A vectorised implementation that multiplies qdata (array) by q1 (single quaternion) """
+
+    w1, x1, y1, z1 = q1
+    w2, x2, y2, z2 = qdata[:,0],qdata[:,1],qdata[:,2],qdata[:,3]
+    w = w1*w2 - x1*x2 - y1*y2 - z1*z2
+    x = w1*x2 + x1*w2 + y1*z2 - z1*y2
+    y = w1*y2 + y1*w2 + z1*x2 - x1*z2
+    z = w1*z2 + z1*w2 + x1*y2 - y1*x2
+    return np.array([w, x, y, z]).T
+
+def _get_rotation_to_beam_direction(beam_direction):
+    """ A helper function for getting rotations around a beam direction, the
+    returns the first two angles (szxz) needed to place the viewer looking down the
+    given zone axis.
+
+    Parameters
+    ----------
+    beam_direction : [vx,vy,vz]
+
+    Returns
+    -------
+    alpha,beta : angles in degrees
+
+    See Also
+    --------
+    generators.get_grid_around_beam_direction
+    """
+    from transforms3d.euler import axangle2euler
+    beam_direction = np.divide(beam_direction,np.linalg.norm(beam_direction))
+    axis = np.cross(beam_direction,[0,0,1]) # [0,0,1] is the starting direction for diffsims
```

Comment:
I don't think this works for non-cubic systems? Surely you need the crystal metric tensor somewhere?